source/particle/world.cc

```diff
@@ -1443,7 +1460,9 @@ namespace aspect
     {
       aspect::oarchive oa (os);
       oa << (*this);
-      output->save(os);
+
+      for ( typename std::vector<std_cxx11::shared_ptr<Output::Interface<dim> > >::const_iterator itr= output.begin(); itr != output.end(); itr++)
+        itr->get()->save(os);
```

Comment:
this, and the corresponding `load` function only works if the list of output formats is exactly the same in the original run as in a restarted run. This is too brittle. Please copy the corresponding code from `postprocess/interface.cc`.